### PR TITLE
feat: Announcement post for transformer embeddings

### PR DESCRIPTION
**Pull request by setu4993**
Here we go!

### CODE REVIEW

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+layout: post
+title: "Open Sourcing Transformer Embeddings"
+date: 2022-10-28
+categories: [nlp, data science, open source]
+author: Setu Shah
+---
+
+The Data Science team at Headspace Health is a heavy user of building, training and productionizing transformer-based NLP models.
+
+Depending on the application, our NLP models operate at varying levels of atomicity: token, word, phrase, sentence, paragraph and document. A lot of our models rely on embeddings from off-the-shelf [transformers models](https://huggingface.co/models), while some are fine-tuned for the application we are building for. Similarly, as a part of our work, we often try out different models and during exploration, while comparing various pooling methods.
```
Comment (jaymeszm) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]:
```suggestion
Depending on the application, our NLP models operate at varying levels of atomicity: token, word, phrase, sentence, paragraph and document. A lot of our models rely on embeddings from off-the-shelf [Transformer models](https://huggingface.co/models), while some are fine-tuned for the application we are building for. Similarly, as a part of our work, we often try out different models during exploration, while comparing various pooling methods.
```

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- Use your model on CPUs or GPUs, without worrying about if you asked PyTorch to use the right device.
+- Export the model and additional artifacts (custom scikit-learn / tree-based models, model cards, etc.) to S3.
+- Customize batch for different models as you play with them.
+
+## You should _not_ use this if you want to...
+
+- Fine-tune the underlying embedding models or train new models. (We recommend [HF `transformers`](https://github.com/huggingface/transformers) or [`sentence-transformers`](https://github.com/UKPLab/sentence-transformers) as alternatives.)
+- Use TensorFlow / JAX for your deep learning models.
+
+`transformer-embeddings` is a battle-tested package for generating embeddings from transformer models in 10+ of our internal production workflows, running on CPUs and GPUs, for real-time and batch inference. We hope you use it, like it and it makes it easier for you to use transformer models, as it has for us :).
```
Comment (jaymeszm) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]: Do we want to add a questions/suggestions section and include a contact?

**Review by jaymeszm — COMMENTED** [review]
🙌🚀 Great post, Setu! Have a few small copy edits, but otherwise this looks great to me.

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown (hunk 2 above)
Comment (setu4993) [reply _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]: I was thinking not.

But just noticed it is on the other ~blogs~ posts, so added it. Good call out!

**Review by jaymeszm — APPROVED** [review]

**Review by akhilchats — APPROVED** [review]
👏

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Today, we are [open-sourcing this package](https://github.com/ginger-io/transformer-embeddings/) and [publishing it to PyPI](https://pypi.org/project/transformer-embeddings/) to make it easy for anyone in the transformer and NLP communities to be able to use aggregated and pooled embeddings for phrases, sentences and paragraphs and documents, at varying levels of aggregation, in any of their applications.
+
+### You should use this if you want to...
+
+- Automatically apply tokenization (with the model defaults) before your model's forward pass.
+- Stack outputs from the model into a single, iterable array that map 1:1 with your input.
+- Simplify interactions with any transformer model available on the [HuggingFace Model Hub](https://huggingface.co/models) for exploration and inference.
+- Easily compare the impact of and then apply different pooling strategies (mean, max, min, pooler) on your downstream tasks.
```
Comment (nxbrnt) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]:
```suggestion
- Easily apply and compare the impact of different pooling strategies (mean, max, min, pooler) on your downstream tasks.
```

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+- Use your model on CPUs or GPUs, without worrying about if you asked PyTorch to use the right device.
+- Export the model and additional artifacts (custom scikit-learn / tree-based models, model cards, etc.) to S3.
+- Customize batch for different models as you play with them.
```
Comment (nxbrnt) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]: "batch size" maybe?

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+## You should _not_ use this if you want to...
+
+- Fine-tune the underlying embedding models or train new models. (We recommend [HF `transformers`](https://github.com/huggingface/transformers) or [`sentence-transformers`](https://github.com/UKPLab/sentence-transformers) as alternatives.)
+- Use TensorFlow / JAX for your deep learning models.
+
+`transformer-embeddings` is a battle-tested package for generating embeddings from transformer models in 10+ of our internal production workflows, running on CPUs and GPUs, for real-time and batch inference. We hope you use it, like it and it makes it easier for you to use transformer models, as it has for us :).
```
Comment (nxbrnt) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]:
```suggestion
`transformer-embeddings` is a battle-tested package for generating embeddings from transformer models in 10+ of our internal production workflows, running on both CPUs and GPUs for real-time and batch inference. We hope you use it, like it, and it makes it easier for you to use transformer models as it has for us :).
```

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+## Questions or suggesstions?
```
Comment (nxbrnt) [inline _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]:
```suggestion
## Questions or suggestions?
```

**Review by nxbrnt — APPROVED** [review]
This looks great, Setu! Added a few suggested edits. Feel free to accept them or not :)

File: _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown (hunk 5 above)
Comment (setu4993) [reply _posts/2022-10-31-open-sourcing-transformer-embeddings.markdown]: Keeping both of those commas.

First is intentional, to separate out the clause for types of instances from the rest of the sentence. Second is, well, an Oxford comma :).

**Pull request merged**